integration_tests/base_routes.py

```diff
@@ -41,239 +39,412 @@ def message():
     return "Hello world, from ws"
 
 
-callCount = 0
+# ===== Lifecucle handlers =====
+
+
+async def startup_handler():
+    print("Starting up")
+
+
+@app.shutdown_handler
+def shutdown_handler():
+    print("Shutting down")
+
+
+# ===== Middlewares =====
+
+
+@app.before_request("/")
+async def hello_before_request(request):
+    request["headers"]["before"] = "before_request"
+    return request
+
+
+@app.after_request("/")
+async def hello_after_request(request):
+    request["headers"]["after"] = "after_request"
+    return request
 
 
 @app.get("/")
-async def hello(request):
-    global callCount
-    callCount += 1
-    message = "Called " + str(callCount) + " times"
-    print(message, request)
-    return {"status_code": 200, "body": "hello", "type": "text"}
+async def middlewares():
+    return ""
+
```

Comment:
Same question - Are we using these?

Reply:
Yes, the new test in `test_middlewares.py` uses this route! It doesn't work for now but will hopefully soon be fixed